client/question/services/FeedbackGeneratorService.js

```diff
@@ -268,28 +360,45 @@ tie.factory('FeedbackGeneratorService', [
      * @private
      */
     var _getCorrectnessTestFeedback = function(
-      outputFunctionName, testCase, observedOutput) {
+      outputFunctionName, testCase, observedOutput, testSuiteId,
+      testSuiteIndex) {
       var allowedOutputExample = testCase.getAnyAllowedOutput();
       var feedback = FeedbackObjectFactory.create(
         FEEDBACK_CATEGORIES.INCORRECT_OUTPUT_FAILURE);
-      // TODO(eyurko): Add varied statements for when code is incorrect.
-      feedback.appendTextParagraph('Your code produced the following result:');
-      if (outputFunctionName) {
-        feedback.appendCodeParagraph(
-          'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
-          ('Output (after running ' + outputFunctionName + '): ' +
-            _jsToHumanReadable(observedOutput)));
-        feedback.appendTextParagraph(
-          'However, the expected output of ' + outputFunctionName + ' is:');
+      if (testSuiteIndex == lastTestSuiteIndex) {
+        // Same test suite; attempt number > 1
+        attemptCounter[testSuiteId] += 1;
+        if (attemptCounter[testSuiteId] == 2 && testSuiteId != 'SAMPLE_INPUT') {
```

Comment:
Don't special case on SAMPLE_INPUT; this should be generic. If buggy/suite-level feedback exists for SAMPLE_INPUT, it will show up prior to correctness feedback being invoked anyway. So there's no need to special-case it here.

Reply:
Per discussion, make this a global constant.